### PR TITLE
Add the Serializer module

### PR DESCRIPTION
**Pull request by hewison-chris**
This adds the Serializer to the crypto submodule so we can conveniently use it in our hashing inplementation in this submodule's `agora.crypto.Hash`.

### CODE REVIEW

**Comment by Geod24** [comment]
Maybe bring https://github.com/bpfkorea/agora/blob/d4248935cbeebcc5a6fa3f845a489852a2ff7c5d/source/agora/test/Crypto.d#L38-L57 back then ?

**Comment by hewison-chris** [comment]
> Maybe bring https://github.com/bpfkorea/agora/blob/d4248935cbeebcc5a6fa3f845a489852a2ff7c5d/source/agora/test/Crypto.d#L38-L57 back then ?

Done

File: source/agora/crypto/Types.d [hunk 1]
```diff
@@ -42,3 +42,10 @@ public enum PrintMode
     /// Print original value
     Clear,
 }
+
+/// Whether integers are serialized in variable-length form
+public enum CompactMode : bool
+{
+    No,
+    Yes
```
Comment (Geod24) [inline source/agora/crypto/Types.d:50]:
```suggestion
    Yes,
```

**Review by Geod24 — APPROVED** [review]

File: source/agora/crypto/Types.d (hunk 1 above)
Comment (hewison-chris) [reply source/agora/crypto/Types.d:50]: As `yes` or `no` enum it is unlikely to ever get more options so I'll leave this time.

Comment (Geod24) [reply source/agora/crypto/Types.d:50]: https://thedailywtf.com/articles/What_Is_Truth_0x3f_

**Pull request merged**